Vagrantfile

```diff
@@ -319,23 +321,41 @@ Vagrant.configure("2") do |config|
         host.hostsupdater.aliases = info[:alias]
       end
 
-      host.vm.provider :virtualbox do |vb|
-        vb.customize ["modifyvm", :id, "--cpus", info[:cpus]]
-        vb.customize ["modifyvm", :id, "--hwvirtex", "on"]
-        vb.customize ["modifyvm", :id, "--memory", info[:mem]]
-        vb.customize ["modifyvm", :id, "--name", hostname]
-        vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
-        vb.customize ["modifyvm", :id, "--natdnshostresolver2", "on"]
-        vb.gui = info[:show_gui]
-        vb.name = hostname
+      if(vagrant_provider == 'virtualbox')
+        host.vm.provider :virtualbox do |vb|
+          vb.customize ["modifyvm", :id, "--cpus", info[:cpus]]
+          vb.customize ["modifyvm", :id, "--hwvirtex", "on"]
+          vb.customize ["modifyvm", :id, "--memory", info[:mem]]
+          vb.customize ["modifyvm", :id, "--name", hostname]
+          vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
+          vb.customize ["modifyvm", :id, "--natdnshostresolver2", "on"]
+          vb.gui = info[:show_gui]
+          vb.name = hostname
+        end
+      elsif(vagrant_provider == 'libvirt')
+        host.vm.provider :libvirt do |libvirt|
```

Comment:
In virtualbox, we ask the VMs to use the host DNS resolver (by enabling `natdnshostresolver1` and `natdnshostresolver2`).

What about libvirt? Which DNS resolver are we using in this case?

Reply:
libvirt runs dnsmasq by default for DHCP and DNS services inside NAT virtual networks,
a dnsmasq instance for each virtual network